src/components/implementation/no_interface/scheddev/sl_mod_fprr.c

```diff
@@ -70,10 +72,32 @@ sl_mod_thd_delete(struct sl_thd_policy *t)
 void
 sl_mod_thd_param_set(struct sl_thd_policy *t, sched_param_type_t type, unsigned int v)
 {
-	assert(type == SCHEDP_PRIO && v < SL_FPRR_NPRIOS);
-	ps_list_rem_d(t); 	/* if we're already on a list, and we're updating priority */
-	t->priority = v;
-	ps_list_head_append_d(&threads[t->priority], t);
+	switch (type) {
+	case SCHEDP_PRIO:
+	{
+		assert(v < SL_FPRR_NPRIOS);
+		ps_list_rem_d(t); 	/* if we're already on a list, and we're updating priority */
+		t->priority = v;
+		ps_list_head_append_d(&threads[t->priority], t);
+		sl_thd_setprio(sl_mod_thd_get(t), t->priority);
+
+		break;
+	}
+	case SCHEDP_WINDOW:
```

Comment:
Is there any way that we can separate the scheduling policy from the timeout stuff, or are they by their nature intertwined?